django/db/models/sql/query.py

```diff
@@ -2052,6 +2052,24 @@ def extra_select(self):
         else:
             return self.extra
 
+    def is_alias_available_join(self, all_paths, alias):
+        """
+        Check alias for whether both from and to in all_paths.
+        If so shouldn't be removed
+        alias_refcount might be changed hence need joined_alias
+        to check available from join
+        """
+        joined_alias = {k for k, v in self.alias_refcount.items() if v > 0}
+        from_to_tables = {
+            path.from_opts.db_table for path in all_paths
+            if path.from_opts.db_table in {
+                path.to_opts.db_table for path in all_paths
+                if path.from_opts.db_table in joined_alias and
+                path.from_opts.db_table != path.to_opts.db_table
+            }
+        }
+        return alias not in from_to_tables
```

Comment:
This seems overly complicated and probably also wrong with tables that are JOIN'ed more than once as the following JOINs will have aliases that are not `db_table`.

Also you probably don't need to build the `from_to_tables` set as it's really inneficent. You should use a loop and exit the function as soon as a matching alias is found.